### PR TITLE
RM - update derived table po_lines_eresource / LDP

### PR DESCRIPTION
**Pull request by adoerrer**
This will add holding_id and relying locations resolving on newly introduced holding information at po line level for derived table po_lines_eresource

### CODE REVIEW

File: sql/derived_tables/po_lines_eresource.sql [hunk 1]
```diff
@@ -18,12 +17,22 @@ WITH temp_eresource AS (
         json_extract_path_text(data, 'eresource', 'userLimit') AS pol_user_limit,
         json_extract_path_text(data, 'eresource', 'resourceUrl') AS pol_resource_url
     FROM
-        po_lines AS pol
+        po_lines AS pol WHERE json_extract_path(data, 'eresource') IS NOT NULL
 )
```
Comment (stdombek) [inline sql/derived_tables/po_lines_eresource.sql:21]: **row 13**
===============
`json_extract_path_text(data, 'eresource', 'license') AS pol_license`
documentation
` "license": {
            "code": "Code 1",
            "description": "License Agreement 1"
          },`
The documentation refers to subfields here. On line 14, the description is accessed as a subfield. In line 13 everything. Do you mean just the subfiled "code" here?

**row 20**
===============
Please put the WHERE clause on a new line and then the condition on a new line with indentation.

File: sql/derived_tables/po_lines_eresource.sql [hunk 2]
```diff
@@ -5,7 +5,6 @@ CREATE TABLE po_lines_eresource AS
 WITH temp_eresource AS (
```
Comment (stdombek) [inline sql/derived_tables/po_lines_eresource.sql:5]: According to the new directive, the documentation line should be at the beginning. Please modify.

File: sql/derived_tables/po_lines_eresource.sql [hunk 3]
```diff
@@ -18,12 +17,22 @@ WITH temp_eresource AS (
...
 SELECT
     te.pol_id,
-    te.pol_location_id,
-    il.name AS location_name,
+    json_extract_path_text(locations.data, 'holdingId') AS pol_holding_id,
+    ih.hrid AS pol_holding_hrid,
+    CASE WHEN json_extract_path_text(locations.data, 'locationId') IS NOT NULL
+	    THEN json_extract_path_text(locations.data, 'locationId') 
+		ELSE ih.permanent_location_id END AS pol_location_id,
+	CASE WHEN (il.name) IS NOT NULL THEN il.name
```
Comment (stdombek) [inline sql/derived_tables/po_lines_eresource.sql:29]: Attribute is surrounded by brackets. The condition should be on a single line and the case distinction on an extra line.

**Review by stdombek — CHANGES_REQUESTED** [review]
- [X] Pull request has a complete description
- [X] Query runs without errors
- [X] Query output is correct
- [X] Query logic is clear and well documented
- [ ] Query is readable and properly indented
- [X] Table and column names are in all-lowercase
- [X] Quotation marks are used only where necessary
- [ ] Query begins with user documentation in comment lines
- [X] File name is listed in `runlist.txt` after dependencies
- [X] All columns have indexes
- [X] Table is vacuumed and analyzed

Looks good, but there are a few things to do.

File: sql/derived_tables/po_lines_eresource.sql (hunk 3 above)
Comment (adoerrer) [reply sql/derived_tables/po_lines_eresource.sql:29]: Have changed that.

File: sql/derived_tables/po_lines_eresource.sql (hunk 2 above)
Comment (adoerrer) [reply sql/derived_tables/po_lines_eresource.sql:5]: Have changed that.

**Review by stdombek — APPROVED** [review]
- [X] Pull request has a complete description
- [X] Query runs without errors
- [X] Query output is correct
- [X] Query logic is clear and well documented
- [X] Query is readable and properly indented
- [X] Table and column names are in all-lowercase
- [X] Quotation marks are used only where necessary
- [X] Query begins with user documentation in comment lines
- [X] File name is listed in `runlist.txt` after dependencies
- [X] All columns have indexes
- [X] Table is vacuumed and analyzed

- Indentation is weird at times, but ok.
- Description in line 1: "local table"? It's probably a matter of taste

**Comment by adoerrer** [comment]
>     * Indentation is weird at times, but ok.
> 
>     * Description in line 1: "local table"? It's probably a matter of taste

Thanks, have fixed this as well. :)

**Review by stdombek — APPROVED** [review]

**Review by natalyapik — COMMENTED** [review]
please delete line 21, 22.

File: sql/derived_tables/po_lines_eresource.sql [hunk 4]
```diff
@@ -1,37 +1,61 @@
...
+        json_extract_path_text(pol.data, 'eresource', 'createInventory') AS pol_create_inventory,
+        json_extract_path_text(pol.data, 'eresource', 'expectedActivation') AS pol_expected_activation,
+        json_extract_path_text(pol.data, 'eresource', 'license', 'code') AS pol_license_code,
+        json_extract_path_text(pol.data, 'eresource', 'license', 'description') AS pol_license_desc,
+        json_extract_path_text(pol.data, 'eresource', 'license', 'reference') AS pol_license_reference,
+        json_extract_path_text(pol.data, 'eresource', 'materialType') AS pol_material_type,
+        json_extract_path_text(pol.data, 'eresource', 'trial') AS pol_trial,
+        json_extract_path_text(pol.data, 'eresource', 'userLimit') AS pol_user_limit,
+        json_extract_path_text(pol.data, 'eresource', 'resourceUrl') AS pol_resource_url,
+        json_extract_path_text(locations.data, 'holdingId') AS pol_holding_id,
```
[the remark this answers is not in the document]
Comment (adoerrer) [reply sql/derived_tables/po_lines_eresource.sql:21]: Hi @natalyapik, from what I understand is that a location entity in POL could be a location_id or an holding_id and these lines should extract holding informations on po lines. 
Unfortunately there are no test examples in snapshot thus far so the columns are empty. I'm not sure how to create an example but do you or perhaps @scotttperry know?

**Comment by natalyapik** [comment]
after talking with Axel I figured it out. This looks great now.

**Review by natalyapik — APPROVED** [review]
looks great!

**Pull request merged**